File: 04-04-2024_Sum_Of_All_Substrings_Of_A_Number.cpp
class Solution{
    public:
    int mod = 1e9+7;
    long long sumSubstrings(string s)
    {
        int n= s.size();
        vector<long long> dp(n,0);
        dp[0] = s[0]-'0';
        long long ans = dp[0];
        for(int i=1;i<n;i++)
        {
            dp[i] = ((dp[i-1]*10)%mod + ((s[i]-'0')*(i+1))%mod)%mod;
            ans = (ans + dp[i])%mod;
        }
        return ans;
    }
};

File: 04-04-2024_Sum_Of_All_Substrings_Of_A_Number.py
class Solution:
    def sumSubstrings(self, s: str) -> int:
        MOD = 10**9 + 7
        n = len(s)
        total_sum = int(s[0])
        current_value = int(s[0])
        
        for i in range(1, n):
            current_value = (current_value * 10 + int(s[i]) * (i + 1)) % MOD
            total_sum = (total_sum + current_value) % MOD
        
        return total_sum
